File: Array/Pow.java
public class Pow {

    public double myPow(double x, int n) {
        Long longN = Long.valueOf(n);
        return calculate(x, longN);
    }

    private double calculate(double x, long n) {
        // base case: if n is 0, return 1
        if (n == 0) {
            return 1;
        }
        // if n is negative, recursively call calculate with 1/x and -n
        if (n < 0) {
            return calculate(1 / x, -n);
        }
        // if n is even, recursively call calculate with x*x and n/2
        if (n % 2 == 0) {
            return calculate(x * x, n / 2);
        }
        // if n is odd, recursively call calculate with x*x and (n-1)/2, then multiply
        // the result by x
        return x * calculate(x * x, (n - 1) / 2);
    }

}

File: Array/Neetcode/Pow.py
class Pow:
    def myPow(self, x, n):
        if n == 0:
            return 1
        elif n < 0:
            return 1 / self.calculate(x, -n)
        else:
            return self.calculate(x, n)

    def calculate(self, x, n):
        if n == 0:
            return 1
        elif n % 2 == 0:
            return self.calculate(x * x, n // 2)
        else:
            return x * self.calculate(x * x, (n - 1) // 2)

# Example Usage:
# pow_instance = Pow()
# result = pow_instance.myPow(2.0, 10)
# print("Result:", result)
